drivers/include/mfrc522.h

```diff
@@ -0,0 +1,844 @@
+                                                        uint8_t *back_data,
+                                                        uint8_t *back_len,
+                                                        uint8_t *valid_bits,
+                                                        uint8_t rx_align,
+                                                        bool check_crc);
+
+/**
+ * @brief Transmits REQA, Type A. Invites PICCs in state IDLE to go to READY and
+ *        prepare for anti-collision or selection. 7 bit frame.
+ *
+ * @note  When two PICCs are in the field at the same time we often get
+ *        MFRC522_STATUS_TIMEOUT - probably due do bad antenna design.
+ *
+ * @param[in]      dev          Device descriptor of the MFRC522
+ * @param[out]     buffer_atqa  Buffer to store the ATQA in
+ * @param[inout]   buffer_size  Buffer size, at least two bytes. Also number of
+ *                              bytes returned if MFRC522_STATUS_OK.
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_picc_request_a(mfrc522_t *dev,
+                                             uint8_t *buffer_atqa,
+                                             uint8_t *buffer_size);
+
+/**
+ * @brief Transmits WUPA, Type A. Invites PICCs in state IDLE and HALT to go to
+ *        READY(*) and prepare for anti-collision or selection. 7 bit frame.
+ *
+ * @note  When two PICCs are in the field at the same time we often get
+ *        MFRC522_STATUS_TIMEOUT - probably due do bad antenna design.
+ *
+ * @param[in]      dev          Device descriptor of the MFRC522
+ * @param[out]     buffer_atqa  Buffer to store the ATQA in
+ * @param[inout]   buffer_size  Buffer size, at least two bytes. Also number of
+ *                              bytes returned if MFRC522_STATUS_OK.
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_picc_wakeup_a(mfrc522_t *dev,
+                                            uint8_t *buffer_atqa,
+                                            uint8_t *buffer_size);
+
+/**
+ * @brief Transmits REQA (Type A) or WUPA (Type A) commands
+ *
+ * @note  When two PICCs are in the field at the same time we often get
+ *        MFRC522_STATUS_TIMEOUT - probably due do bad antenna design.
+ *
+ * @param[in]     dev          Device descriptor of the MFRC522
+ * @param[in]     command      Command to send - MFRC522_PICC_CMD_ISO_14443_REQA
+ *                             or MFRC522_PICC_CMD_ISO_14443_WUPA
+ * @param[out]    buffer_atqa  Buffer to store the ATQA in
+ * @param[inout]  buffer_size  Buffer size, at least two bytes. Also number of
+ *                             bytes returned if MFRC522_STATUS_OK.
+ *
+ * @return MFRC522_STATUS_OK on success
+ * @return MFRC522_STATUS_INVALID if command was neither
+ *         MFRC522_PICC_CMD_ISO_14443_REQA nor MFRC522_PICC_CMD_ISO_14443_WUPA,
+ *         MFRC522_STATUS_??? otherwise
```

Comment:
We could change this a bit. Add a `@note` with valid commands in the brief, and just say `@retval MFRC522_STATUS_INVALID if command is not valid`. Though I guess this will change if switched to errno.